quint/src/static/callgraph.ts

```diff
@@ -167,42 +176,49 @@ export class CallGraphVisitor implements IRVisitor {
     this.stack.pop()
   }
 
-  enterImport(def: QuintImport) {
-    const importedModule = this.context.modulesByName.get(def.protoName)
+  enterImport(decl: QuintImport) {
+    const importedModule = this.context.modulesByName.get(decl.protoName)
     if (importedModule) {
-      this.graphAddAll(def.id, Set([importedModule.id]))
+      this.graphAddAll(decl.id, Set([importedModule.id]))
     }
   }
 
-  enterInstance(def: QuintInstance) {
-    const importedModule = this.context.modulesByName.get(def.protoName)
+  enterInstance(decl: QuintInstance) {
+    // Instances are the only non-definition declarations that need to be added to the stack,
+    // because they may contain names in the overrides
+    this.stack.push(decl)
+    const importedModule = this.context.modulesByName.get(decl.protoName)
     if (importedModule) {
-      this.graphAddAll(def.id, Set([importedModule.id]))
+      this.graphAddAll(decl.id, Set([importedModule.id]))
     }
   }
 
-  enterExport(def: QuintExport) {
-    const key = mkImportKey(this.currentModuleId, def.protoName)
+  exitInstance(_decl: QuintInstance) {
+    this.stack.pop()
+  }
```

Comment:
@konnov can you take a look at this part? I could make it work, but I'm not 100% sure it is sound.

Basically, the changes impacting this are:
1. This uses `exitDef`, which no longer include imports/exports/instances
2. If we change it to `exitDecl`, it is not the same as before, since a nested definition is not a declaration (that is, opdefs inside `QuintLet`).

So I kept `exitDef` and added these `enterInstance` and `exitInstance` to handle dependencies coming from expressions in overrides. IIUC, there is no need to do the same for imports and exports.

Generally speaking, I can see there is a small overhead, in some cases, arising from the fact that we no longer have an `exitSomething` that includes both imports/exports/instances and nested defs. But for the most cases, I think this is a good thing, and it is helpful to have explicitly different treatment for declarations and definitions.

Reply:
> Is it that all the definitions inside an instance depends on the instance name
I don't think we need this. 

> Given that, I don't think that push and pop are doing much, except maybe adding the dependencies on the instance parameters
I think that's what I was aiming at, so something like `import A(N = bar).*` has a dependency on `bar`.